Use bang as force write

lukas-reineke opened the pull request:
Only write format output to the buffer when there are no unsaved changes or `Format` was called with bang

I added `force` as a new param for now, but when we want to add more stuff, it'll probably be nicer to have a config table or similar.

mhartington commented:
With the latest changes to refactor branch, could you update and we could merge this in?

lukas-reineke commented:
Updated
and I changed some other things too

1. we read the buffer only once and keep the state in memory between formatter, then write back at the end.
	We need to do this, because one formatter can change the number of lines, which then will make the next formatter read the wrong selection from the buffer. Or will throw an `Index out of bounds` error on write
2. only read the bufnr once, so the user can focus a different buffer and things still work
3. I removed the `getRange` function. `-range=%` will always set the start and end line, they will never be `nil`.
4. I merged `format` and `internalFormatter` as there was basically no logic left in `format`. Is there a reason to keep it?

mhartington commented:
Looks good!
